### DISCUSSION TITLE
計算ノードで待機中から進まない件について

### DISCUSSION THREAD

**Discussion by ikumi1220** [nig-sc/nigsc_homepage2#374] [Q&A] [answered]
ディレクトリ内のファイルをhisat2で処理するためにforコマンドを用いたシェルスクリプトを作成し、qsubコマンドでジョブを計算ノードに投入しました。しかし、24時間以上経過したのにもかかわらず待機中のまま進行していません。
`$ qstat -g c`
で確認したところ、利用可能なスロットは十分にあいているように思うのですが、待機中から進展しない理由が特定できません。
スクリプトに問題があれば教えていただけると幸いです。

問題のシェルスクリプトはこちらです。

```
#!/bin/bash

#$ -l s_vmem=32G
#$ -l mem_req=32G
#$ -V
#$ -S /bin/bash
#$ -cwd

files=($(ls /home/ikumi1220/trimmedfastq))

for ((i = 0; i < ${#files[@]}; i += 2)); do
file1="${files[$i]}"
file2="${files[$i + 1]}"

/opt/pkg/singularity-ce/3.10.2/bin/singularity exec /usr/local/biotools/h/hisat2:2.2.1--py38he1b5a44_0 \
hisat2 -q -x /home/ikumi1220/ref/grch38/genome -1 /home/ikumi1220//trimmedfastq/"$file1" -2 /home/ikumi1220/trimmedfastq/"$file2" -S ~/mapped/"$file1".sam

done

```

また、hisat2で問題が発生している可能性を考え、forコマンドを使用せずにqsubで以下のhisat2のスクリプトを投入したところ正常に動作しました。

```
#!/usr/bin/bash

#$ -l s_vmem=16G
#$ -l mem_req=16G

/opt/pkg/singularity-ce/3.10.2/bin/singularity exec /usr/local/biotools/h/hisat2:2.2.1--py38he1b5a44_0 \
hisat2 -q -x ~/ref/grch38/genome -1 ~/trimmedfastq/fastq1.fq.gz -2 ~/trimmedfastq/fastq2.fq.gz -S ~/mapped/mapped.sam

```

加えて、予備的に上のforコマンドを用いたスクリプトを計算ノードで動作させた際、hisat2の要求メモリーが足りなかったため正常に動作はしませんでしたが、hisat2がrunするところまでは動作しました。

qsubでこのスクリプトを投入した際以下の文が表示されるのですが、関係ありますでしょうか。

`removed environment variable LD_LIBRARY_PATH_modshare from submit environment - it is considered a security issue`

初心者故基本的な質問で申し訳ありません。よろしくお願いします。

**Comment by sc-fs** [comment, marked as answer]
待ち時間が長く申し訳ありまりません。

qstat -g c　の結果でslotは空きがありように見えますが、
JOBの実行にはメモリも必要となりますので、こちらに空きがなければ、JOBが実行できません。
その他、複数の方での共同利用となりますので、他にＪＯＢがあれば、待つことになります。
現在、ikumi1220様の待ちＪＯＢ：24680137の前に約100JOB待っている状況です。
以下に各キューのJOB蓄積状況があります。
https://sc.ddbj.nig.ac.jp/operation/
ご迷惑をお掛けしますが、もうしばらくお待ちください。

提示して頂いた、スクリプト及び、投入オプションには問題は見られません。
もし、処理時間が１時間以内で済むようであれば、shortキューへの投入を検討ください。
（処理時間が１時間を超えると、強制的に処理が中断します）

`qsub -l short /home/ikumi1220/script/hisat2.sh`


forコマンドを取り除いたJOBには「#$ -S /bin/bash」の指定がありません。また、「#!/usr/bin/bash」の指定も先頭のスクリプトと同じく「#!/bin/bash」にしていただけますでしょうか。
